src/App.tsx

```diff
@@ -14,11 +16,100 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[],
+};
+
+type Props = {};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  goodsArr = this.state.selectedGood;
```

Comment:
on textMaker you receive goodsArr as props. When you call textMaker you use selectedGood:
`this.textMaker(this.state.selectedGood)`
so I think that you can delete goodsArr here

```suggestion
```